### ISSUE TITLE
Undo (CTRL+Z) reverses EOL conversion but is not updated on UI

### ISSUE THREAD

**Issue by juanpenia** [notepad-plus-plus/notepad-plus-plus#14249]
### Steps to Reproduce the Issue

1. Have a file with EOL on CR LF.
2. Convert to LF using Edit > EOL Conversion > Unix (LF).
3.  Save with CTRL+S or whatever.
4. Undo with CTRL+Z.

### Expected Behavior

Show on the status bar below the correct EOL setting.

### Actual Behavior
Shows the previous EOL setting.

### Debug Information
Notepad++ v8.5.8   (64-bit)
Build time : Oct 15 2023 - 21:43:56
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "N:\iso2023\4.sh" 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 11 Pro (64-bit)
OS Version : 22H2
OS Build : 22621.2428
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)

video proof:
[https://youtu.be/hIVmYWs4Brc](url)

**Comment by mpheath** [comment]
Working link: https://youtu.be/hIVmYWs4Brc

The document changed with Ctrl+Z (undo) the EOL sequences from **LF** to **CRLF**, though the EOL setting is still **LF**. If you type on a new line some text and press the Enter key, the EOL character will be **LF**, not **CRLF**. This will be the same as the value in the statusbar.

The value of EOL in the statusbar reflects the sequence of **LF**, **CR** or **CRLF** that is inserted when the Enter key occurs.

You seem to consider that the undo event changes the EOL mode setting? It does not, it only undoes the changes in the document.

If the PythonScript plugin is installed, type `['CRLF', 'CR', 'LF'][editor.getEOLMode()]` into the Run box and click the Run button and should see printed `LF` which matches the statusbar EOL value of `LF`.

The statusbar EOL value appears to be correct.

**Comment by alankilborn** [comment]
I agree with @mpheath's assessment.  "Undo" undoes textual content changes; it does not allow you to walk back through the _command_ list executed by N++.

The correct way to undo a line ending conversion operation is to simply do another one.

**Comment by juanpenia** [comment]
To be honest, at least what seems to happen on a visual level is that when I undo, it did reverse the EOL conversion since you see nothing is changed in the code. But yes, I see your point and could be just something changing in the code that I don't really see.
